### PR TITLE
feat: enable custom logits processors

### PR DESCRIPTION
**Pull request by wallashss**
# Description

This PR enable to add custom logits processors.

### CODE REVIEW

File: vllm_spyre/v1/worker/spyre_model_runner.py [hunk 1]
```diff
@@ -305,9 +306,17 @@ def load_model(self, prompt_lens: Iterable[int],
...
-        # constructor argument; for now the list is fixed to builtin processors
-        logits_processors = get_builtin_logits_processors(self.vllm_config)
+
+        # TODO: compatibility code, remove it when it's not needed anymore
+        if hasattr(vllm.v1.sample.logits_processor, "LogitsProcessors"):
+            from vllm.v1.sample.logits_processor import build_logitsprocs
+
+            logits_processors = build_logitsprocs(
+                self.vllm_config, self.device, self.pin_memory, False,
+                self.vllm_config.model_config.logits_processors)
```
Comment (maxdebayser) [inline vllm_spyre/v1/worker/spyre_model_runner.py:316]: Nice, but can you update `get_builtin_logits_processors` to use `build_logitsprocs` instead? Maybe we should also rename the method to  `get_logits_processors`.

**Review by maxdebayser — CHANGES_REQUESTED** [review]

File: vllm_spyre/v1/worker/spyre_model_runner.py (hunk 1 above)
Comment (wallashss) [reply vllm_spyre/v1/worker/spyre_model_runner.py]: Thanks for the suggestion! Makes sense. Also, there was a PR that we removed some compatibility code, then we could simplify a lot this implementation.

**Comment by wallashss** [comment]
bot:test
